### PR TITLE
Bugfix when build model with strict=False for load state dict

### PR DESCRIPTION
**Pull request by KonstantinWilleke**

### CODE REVIEW

File: nnfabrik/utility/nn_helpers.py [hunk 1]
```diff
@@ -129,6 +129,7 @@ def load_state_dict(
     match_names: bool = False,
     ignore_dim_mismatch: bool = False,
     prefix_agreement: float = 0.98,
+    strict=None,
```
Comment (MaxFBurg) [inline nnfabrik/utility/nn_helpers.py:132]:
```suggestion
    strict=True,
```

To keep default behavior

File: nnfabrik/utility/nn_helpers.py [hunk 2]
```diff
@@ -157,29 +158,31 @@ def load_state_dict(
...
-    updated_model_dict = {}
-    for k, v in filtered_state_dict.items():
-        if v.shape != model_dict[k].shape:
-            if ignore_dim_mismatch:
-                print("Ignored shape-mismatched parameter:", k)
-                continue
-            else:
-                raise RuntimeError("Error in loading state_dict: Shape-mismatch for key {}".format(k))
-        updated_model_dict[k] = v
+    if strict is None:
```
Comment (MaxFBurg) [inline nnfabrik/utility/nn_helpers.py:161]:
```suggestion
    if strict:
```

File: nnfabrik/utility/nn_helpers.py [hunk 3]
```diff
@@ -157,29 +158,31 @@ def load_state_dict(
...
+                    continue
+                else:
+                    raise RuntimeError("Error in loading state_dict: Shape-mismatch for key {}".format(k))
+            updated_model_dict[k] = v
+    else:
+        updated_model_dict = state_dict
 
     # 3. load the new state dict
-    model.load_state_dict(updated_model_dict, strict=(not ignore_missing))
+    model.load_state_dict(updated_model_dict, strict=(not ignore_missing) if strict is None else strict)
```
Comment (MaxFBurg) [inline nnfabrik/utility/nn_helpers.py:188]:
```suggestion
    model.load_state_dict(updated_model_dict, strict=(not ignore_missing) if strict is True else False)
```

**Review by MaxFBurg — CHANGES_REQUESTED** [review]

**Comment by MaxFBurg** [comment]
I think it would be cool and helpful to have this feature. I just added a review, @KonstantinWilleke could you have a quick look and commit the changes I requested? (if @ArneNx is fine with me reviewing instead?)

**Comment by ArneNx** [comment]
Thank you for handling this Max. I don't use this anymore, so I'm fine if you take over on this review.

**Review by MaxFBurg — APPROVED** [review]

**Pull request merged**